src/service.rs

```diff
@@ -474,4 +519,34 @@ impl Service {
             Err(io::Error::last_os_error().into())
         }
     }
+
+    /// Get the service config from the system.
+    pub fn query_config(&self) -> Result<ServiceInfo> {
+        let mut data: Vec<u8>;
+        let mut bytes_needed: u32 = 0;
+        unsafe {
+            winsvc::QueryServiceConfigW(
+                self.service_handle.raw_handle(),
+                ::std::ptr::null_mut(),
+                mem::size_of::<winsvc::QUERY_SERVICE_CONFIGW>() as u32,
+                &mut bytes_needed,
+            )
+        };
+        let success = unsafe {
+            data = vec![0; bytes_needed as usize];
+            winsvc::QueryServiceConfigW(
+                self.service_handle.raw_handle(),
+                data.as_mut_ptr() as _,
+                data.len() as u32,
+                &mut bytes_needed,
+            )
+        };
+        if success == 1 {
```

Comment:
It got me thinking that initially the error handling was not correct in the rest of `Service` implementation. As per docs `winsvc::` family of functions return zero on failure, and non-zero code upon success. 

Non-zero means that technically it can return `-1` or `2`. I am pushing a PR to fix it in the rest of API calls that we do in the Service implementation. Basically the solution is to inverse the if-else and check against `0`, i.e:

```rs
if success == 0 {
  Err(…)
} else {
  Ok(…)
}
```

Reply:
i changed the way these are matched, 0 is err, everything else is success.